src/prefect/storage/git.py

```diff
@@ -0,0 +1,149 @@
+    ) -> None:
+        if tag and branch_name:
+            raise ValueError(
+                "Either `tag` or `branch_name` can be specified, but not both"
+            )
+
+        self.flow_path = flow_path
+        self.repo = repo
+        self.repo_host = repo_host
+        self.flow_name = flow_name
+        self.git_token_secret_name = git_token_secret_name
+        self.git_token_username = (
+            git_token_username if git_token_username else repo.split("/")[0]
+        )
+        self.branch_name = branch_name
+        self.tag = tag
+        self.clone_depth = clone_depth
+        self.use_ssh = use_ssh
+        self.format_access_token = format_access_token
+        super().__init__(**kwargs)
+
+    def get_flow(self, flow_name: str) -> "Flow":
+        """
+        Given a flow name within this Storage object, load and return the Flow.
+        Args:
+            - flow_name (str): the name of the flow to return.
+        Returns:
+            - Flow: the requested flow
+        """
+        if flow_name not in self.flows:
+            raise ValueError("Flow is not contained in this Storage")
+
+        with TemporaryGitRepo(
+            git_clone_url=self.git_clone_url,
+            branch_name=self.branch_name,
+            tag=self.tag,
+            clone_depth=self.clone_depth,
+        ) as temp_repo:
+            flow = extract_flow_from_file(
+                file_path=os.path.join(temp_repo.temp_dir.name, self.flow_path),
+                flow_name=self.flow_name,
+            )
+        return flow
+
+    def add_flow(self, flow: "Flow") -> str:
+        """
+        Method for storing a new flow as bytes in the local filesytem.
+        Args:
+            - flow (Flow): a Prefect Flow to add
+        Returns:
+            - str: the location of the added flow in the repo
+        Raises:
+            - ValueError: if a flow with the same name is already contained in this storage
+        """
+        if flow.name in self:
+            raise ValueError(
+                'Name conflict: Flow with the name "{}" is already present in this storage.'.format(
+                    flow.name
+                )
+            )
+        self.flows[flow.name] = self.flow_path
+        self._flows[flow.name] = flow
+        return self.flow_path
+
+    @property
+    def git_token_secret(self) -> str:
+        """
+        Get and format the git secret token if it exists
+        """
+        if self.git_token_secret_name is None:
+            return ""
+
+        # get the access token and format it for common git hosts
+        access_token = Secret(self.git_token_secret_name).get()
+        if self.format_access_token:
+            return f"{self.git_token_username}:{access_token}"
+        return str(access_token)
+
+    @property
+    def git_clone_url(self) -> str:
+        """
+        Build the git url to clone
+        """
+        if self.use_ssh:
+            return f"git@{self.repo_host}:{self.repo}.git"
```

Comment:
Should probably warn if a secret is set here

Reply:
added a warning in __init__, hopefully will allow users to catch it right away